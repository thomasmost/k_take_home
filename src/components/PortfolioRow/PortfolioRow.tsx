import styled from '@emotion/styled';
import * as React from 'react';

interface IPortfolioRowProps {
  denomInfo: DenomInfo;
}

// Some of these styled components are duped;
// could be broken into a shared file

const PillLayout = styled.div`
  background-color: white;
  border: 1px solid lightgray;
  border-radius: 10px;
  display: flex;
  padding: 20px;
  justify-content: start;
  margin-bottom: 10px;
`;


const HeaderLayout = styled.div`
  border: 1px solid transparent;
  border-radius: 10px;
  display: flex;
  padding: 0 20px;
  display: flex;
  justify-content: start;
`;

const AssetCol = styled.div`
  width: 200px;
`;
const TotalCol = styled.div`
  width: 180px;
`;
const AvailableCol = styled.div`
  width: 80px;
`;
const LockedCol = styled.div`
  width: 80px;
`;

const MainData = styled.div`
  font-size: 14px;
`;
const SecondaryData = styled.div`
  margin-top: 4px;
  font-size: 10px;
  color: gray;
`;

export const PortfolioRow: React.FC<IPortfolioRowProps> = ({denomInfo}) => {
  const {denom, displayPrice, totalTokens, totalDollars, lockedDollars, liquidDollars} = denomInfo;
  let displayDenom = denom;
  // Obviously hacky but until I understand it better will do for now
  if (denom === 'ukava') {
    displayDenom = 'kava';
  }
  return (
    <PillLayout>
        <AssetCol>
          <MainData>{displayDenom.toUpperCase()}</MainData>
          <SecondaryData>${displayPrice}</SecondaryData>
        </AssetCol>
        <TotalCol>
          <MainData>
            {totalTokens} {displayDenom.toUpperCase()}
          </MainData>
          <SecondaryData>${totalDollars.toFixed(2)}</SecondaryData>
        </TotalCol>
        <AvailableCol>
          <MainData>
            ${liquidDollars.toFixed(2)}
          </MainData>
        </AvailableCol>
        <LockedCol>
          <MainData>
            ${lockedDollars.toFixed(2)}
          </MainData>
        </LockedCol>
    </PillLayout>
  )
};
