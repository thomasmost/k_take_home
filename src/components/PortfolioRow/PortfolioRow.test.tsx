import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { PortfolioRow } from './PortfolioRow';

const denomInfo= {
  denom: 'kava',
  displayPrice: '4.06',
  totalDollars: 145.49,
  totalTokens: 35.862256,
  liquidDollars: 99.05,
  lockedDollars: 46.44
}

test('PortfolioSection renders', () => {
  const component = renderer.create(
    <PortfolioRow denomInfo={denomInfo} />
  );
  expect(component).toMatchSnapshot();
});
