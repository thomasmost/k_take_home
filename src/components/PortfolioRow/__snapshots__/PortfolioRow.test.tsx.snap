// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioSection renders 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-hh9nq6{background-color:white;border:1px solid lightgray;border-radius:10px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:20px;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:start;justify-content:start;margin-bottom:10px;}",
      }
    }
    data-emotion="css hh9nq6"
  />,
  <div
    className="css-hh9nq6"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1inivuc{width:200px;}",
        }
      }
      data-emotion="css 1inivuc"
    />
    <div
      className="css-1inivuc"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-mr3j8u{font-size:14px;}",
          }
        }
        data-emotion="css mr3j8u"
      />
      <div
        className="css-mr3j8u"
      >
        KAVA
      </div>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-wx0j7l{margin-top:4px;font-size:10px;color:gray;}",
          }
        }
        data-emotion="css wx0j7l"
      />
      <div
        className="css-wx0j7l"
      >
        $
        4.06
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1rfg5hd{width:180px;}",
        }
      }
      data-emotion="css 1rfg5hd"
    />
    <div
      className="css-1rfg5hd"
    >
      <div
        className="css-mr3j8u"
      >
        35.862256
         
        KAVA
      </div>
      <div
        className="css-wx0j7l"
      >
        $
        145.49
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-hscz2w{width:80px;}",
        }
      }
      data-emotion="css hscz2w"
    />
    <div
      className="css-hscz2w"
    >
      <div
        className="css-mr3j8u"
      >
        $
        99.05
      </div>
    </div>
    <div
      className="css-hscz2w"
    >
      <div
        className="css-mr3j8u"
      >
        $
        46.44
      </div>
    </div>
  </div>,
]
`;
